### PR TITLE
optimize initAPIService

### PR DESCRIPTION
**Pull request by cleverhu**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Optimize initAPIService function.Make it more concise.
**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by jwcesign** [comment]
/lgtm
cc @Poor12

**Comment by RainbowMango** [comment]
@cleverhu Cloud you please post the test result with this patch?

**Comment by cleverhu** [comment]
> @cleverhu Cloud you please post the test result with this patch?

Ok,I will do it later.

File: pkg/karmadactl/util/idempotency.go [hunk 1]
```diff
@@ -88,6 +88,8 @@ func CreateOrUpdateAPIService(apiRegistrationClient *aggregator.Clientset, apise
 			return fmt.Errorf("unable to update apiService: %v", err)
 		}
 	}
+
+	klog.V(2).Infof("apiService %s has been created or updated.", apiservice.ObjectMeta.Name)
```
[the remark this answers is not in the document]
Comment (cleverhu) [reply pkg/karmadactl/util/idempotency.go:92]: > @cleverhu 
> 
> 
> 
> `apiservice.Name ` is better than `apiservice.ObjectMeta.Name`

However, all the names under this file are named in this way, and I will propose a pr to unify it.

**Comment by cleverhu** [comment]
> @cleverhu Cloud you please post the test result with this patch?

I test it locally, and it works well.
This is the log for karmadactl.
```
⚡ root@k8s-master  ~/go/src/karmada   optimize-deploy-cmd go run cmd/karmadactl/karmadactl.go init --kube-image-registry=registry.cn-hangzhou.aliyuncs.com/google_containers --etcd-init-image=docker.io/library/alpine:3.15.1 --karmada-scheduler-image=swr.ap-southeast-1.myhuaweicloud.com/karmada/karmada-scheduler:v1.3.0   --crds='https://ghproxy.com/github.com/karmada-io/karmada/releases/download/v1.3.0/crds.tar.gz' --karmada-aggregated-apiserver-image='swr.ap-southeast-1.myhuaweicloud.com/karmada/karmada-aggregated-apiserver:v1.3.0' --karmada-controller-manager-image='swr.ap-southeast-1.myhuaweicloud.com/karmada/karmada-controller-manager:v1.3.0' --karmada-web
hook-image='swr.ap-southeast-1.myhuaweicloud.com/karmada/karmada-webhook:v1.3.0' 
I1117 13:22:37.941434  435363 deploy.go:175] kubeconfig file: , kubernetes: https://10.130.59.18:6443
I1117 13:22:37.964728  435363 deploy.go:195] karmada apiserver ip: [10.130.59.18]
I1117 13:22:38.445949  435363 cert.go:229] Generate ca certificate success.
I1117 13:22:38.573403  435363 cert.go:229] Generate karmada certificate success.
I1117 13:22:38.706810  435363 cert.go:229] Generate apiserver certificate success.
I1117 13:22:38.924902  435363 cert.go:229] Generate front-proxy-ca certificate success.
I1117 13:22:39.148434  435363 cert.go:229] Generate front-proxy-client certificate success.
I1117 13:22:39.355021  435363 cert.go:229] Generate etcd-ca certificate success.
I1117 13:22:39.486328  435363 cert.go:229] Generate etcd-server certificate success.
I1117 13:22:39.671497  435363 cert.go:229] Generate etcd-client certificate success.
I1117 13:22:39.671678  435363 deploy.go:288] download crds file:https://ghproxy.com/github.com/karmada-io/karmada/releases/download/v1.3.0/crds.tar.gz
Downloading...[ 100.00% ]
Download complete.
I1117 13:22:41.380472  435363 deploy.go:524] Create karmada kubeconfig success.
I1117 13:22:41.386548  435363 idempotency.go:192] Namespace karmada-system has been created or updated.
I1117 13:22:41.423188  435363 idempotency.go:215] Service karmada-system/etcd has been created or updated.
I1117 13:22:41.423202  435363 deploy.go:353] create etcd StatefulSets
W1117 13:22:41.470182  435363 check.go:101] etcd desired replicaset is 1, currently: 0
I1117 13:22:42.472613  435363 check.go:98] etcd desired replicaset is 1, currently: 1
I1117 13:22:45.476650  435363 check.go:49] pod: etcd-0 is ready. status: Running
I1117 13:22:45.476668  435363 deploy.go:364] create karmada ApiServer Deployment
I1117 13:22:45.485181  435363 idempotency.go:215] Service karmada-system/karmada-apiserver has been created or updated.
W1117 13:22:48.499726  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:22:49.501943  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:22:50.502434  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:22:51.502103  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:22:52.502438  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:22:53.501601  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:22:54.501811  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:22:55.502258  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:22:56.501505  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:22:57.501787  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:22:58.501736  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:22:59.502072  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:23:00.501886  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:23:01.502272  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:23:02.501715  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:23:03.502403  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:23:04.501289  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:23:05.502223  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:23:06.502768  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:23:07.502042  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:23:08.501737  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:23:09.502330  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:23:10.502223  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:23:11.501525  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:23:12.501582  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:23:13.502220  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:23:14.501675  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
W1117 13:23:15.592291  435363 check.go:52] pod: karmada-apiserver-645bff9854-x2nlk not ready. status: Running
I1117 13:23:16.501516  435363 check.go:49] pod: karmada-apiserver-645bff9854-x2nlk is ready. status: Running
I1117 13:23:16.501542  435363 deploy.go:377] create karmada aggregated apiserver Deployment
I1117 13:23:16.509170  435363 idempotency.go:215] Service karmada-system/karmada-aggregated-apiserver has been created or updated.
I1117 13:23:19.516589  435363 check.go:49] pod: karmada-aggregated-apiserver-5f77d747fd-z5c5s is ready. status: Running
I1117 13:23:19.530924  435363 idempotency.go:192] Namespace karmada-system has been created or updated.
I1117 13:23:19.531062  435363 deploy.go:69] Initialize karmada bases crd resource `/etc/karmada/crds/bases`
I1117 13:23:19.533908  435363 deploy.go:213] Attempting to create CRD
I1117 13:23:19.539633  435363 deploy.go:219] CRD resourceinterpreterwebhookconfigurations.config.karmada.io already exists, skipping.
I1117 13:23:19.540122  435363 deploy.go:213] Attempting to create CRD
I1117 13:23:19.544306  435363 deploy.go:219] CRD serviceexports.multicluster.x-k8s.io already exists, skipping.
I1117 13:23:19.544973  435363 deploy.go:213] Attempting to create CRD
I1117 13:23:19.549104  435363 deploy.go:219] CRD serviceimports.multicluster.x-k8s.io already exists, skipping.
I1117 13:23:19.550379  435363 deploy.go:213] Attempting to create CRD
I1117 13:23:19.556510  435363 deploy.go:219] CRD multiclusteringresses.networking.karmada.io already exists, skipping.
I1117 13:23:19.558683  435363 deploy.go:213] Attempting to create CRD
I1117 13:23:19.566902  435363 deploy.go:219] CRD clusteroverridepolicies.policy.karmada.io already exists, skipping.
I1117 13:23:19.568734  435363 deploy.go:213] Attempting to create CRD
I1117 13:23:19.575830  435363 deploy.go:219] CRD clusterpropagationpolicies.policy.karmada.io already exists, skipping.
I1117 13:23:19.576456  435363 deploy.go:213] Attempting to create CRD
I1117 13:23:19.580816  435363 deploy.go:219] CRD federatedresourcequotas.policy.karmada.io already exists, skipping.
I1117 13:23:19.583144  435363 deploy.go:213] Attempting to create CRD
I1117 13:23:19.590868  435363 deploy.go:219] CRD overridepolicies.policy.karmada.io already exists, skipping.
I1117 13:23:19.592636  435363 deploy.go:213] Attempting to create CRD
I1117 13:23:19.599667  435363 deploy.go:219] CRD propagationpolicies.policy.karmada.io already exists, skipping.
I1117 13:23:19.603476  435363 deploy.go:213] Attempting to create CRD
I1117 13:23:19.612600  435363 deploy.go:219] CRD clusterresourcebindings.work.karmada.io already exists, skipping.
I1117 13:23:19.615057  435363 deploy.go:213] Attempting to create CRD
I1117 13:23:19.743617  435363 deploy.go:219] CRD resourcebindings.work.karmada.io already exists, skipping.
I1117 13:23:19.744463  435363 deploy.go:213] Attempting to create CRD
I1117 13:23:19.940918  435363 deploy.go:219] CRD works.work.karmada.io already exists, skipping.
I1117 13:23:19.940989  435363 deploy.go:80] Initialize karmada patches crd resource `/etc/karmada/crds/patches`
I1117 13:23:20.350221  435363 deploy.go:92] Create MutatingWebhookConfiguration mutating-config.
I1117 13:23:20.359536  435363 webhook_configuration.go:195] MutatingWebhookConfiguration mutating-config already exists, skipping.
I1117 13:23:20.359553  435363 deploy.go:97] Create ValidatingWebhookConfiguration validating-config.
I1117 13:23:20.364116  435363 webhook_configuration.go:175] ValidatingWebhookConfiguration validating-config already exists, skipping.
I1117 13:23:20.370749  435363 idempotency.go:215] Service karmada-system/karmada-aggregated-apiserver has been created or updated.
I1117 13:23:20.378732  435363 check.go:26] Waiting for APIService(v1alpha1.cluster.karmada.io) condition(Available), will try
I1117 13:23:21.381363  435363 check.go:26] Waiting for APIService(v1alpha1.cluster.karmada.io) condition(Available), will try
I1117 13:23:22.381597  435363 check.go:26] Waiting for APIService(v1alpha1.cluster.karmada.io) condition(Available), will try
I1117 13:23:23.381158  435363 check.go:26] Waiting for APIService(v1alpha1.cluster.karmada.io) condition(Available), will try
I1117 13:23:24.381075  435363 check.go:26] Waiting for APIService(v1alpha1.cluster.karmada.io) condition(Available), will try
I1117 13:23:25.381172  435363 check.go:26] Waiting for APIService(v1alpha1.cluster.karmada.io) condition(Available), will try
I1117 13:23:26.381568  435363 check.go:26] Waiting for APIService(v1alpha1.cluster.karmada.io) condition(Available), will try
I1117 13:23:27.381479  435363 check.go:26] Waiting for APIService(v1alpha1.cluster.karmada.io) condition(Available), will try
I1117 13:23:28.381729  435363 check.go:26] Waiting for APIService(v1alpha1.cluster.karmada.io) condition(Available), will try
I1117 13:23:29.381523  435363 check.go:26] Waiting for APIService(v1alpha1.cluster.karmada.io) condition(Available), will try
I1117 13:23:30.381413  435363 check.go:26] Waiting for APIService(v1alpha1.cluster.karmada.io) condition(Available), will try
I1117 13:23:31.381745  435363 check.go:26] Waiting for APIService(v1alpha1.cluster.karmada.io) condition(Available), will try
I1117 13:23:32.380824  435363 check.go:26] Waiting for APIService(v1alpha1.cluster.karmada.io) condition(Available), will try
I1117 13:23:33.381538  435363 check.go:26] Waiting for APIService(v1alpha1.cluster.karmada.io) condition(Available), will try
I1117 13:23:34.381729  435363 check.go:26] Waiting for APIService(v1alpha1.cluster.karmada.io) condition(Available), will try
I1117 13:23:35.381773  435363 check.go:26] Waiting for APIService(v1alpha1.cluster.karmada.io) condition(Available), will try
I1117 13:23:36.381578  435363 check.go:26] Waiting for APIService(v1alpha1.cluster.karmada.io) condition(Available), will try
I1117 13:23:37.383093  435363 rbac.go:65] CreateClusterRole cluster-proxy-admin already exists.
I1117 13:23:37.384643  435363 rbac.go:86] CreateClusterRoleBinding cluster-proxy-admin already exists.
I1117 13:23:37.409491  435363 rbac.go:65] CreateClusterRole system:karmada:agent already exists.
I1117 13:23:37.410812  435363 rbac.go:86] CreateClusterRoleBinding system:karmada:agent already exists.
I1117 13:23:37.410829  435363 tlsbootstrap.go:32] [bootstrap-token] configured RBAC rules to allow Karmada Agent Bootstrap tokens to post CSRs in order for agent to get long term certificate credentials
I1117 13:23:37.583859  435363 rbac.go:86] CreateClusterRoleBinding karmada:agent-bootstrap already exists.
I1117 13:23:37.583876  435363 tlsbootstrap.go:46] [bootstrap-token] configured RBAC rules to allow the csrapprover controller automatically approve CSRs from a Karmada Agent Bootstrap Token
I1117 13:23:37.783600  435363 rbac.go:86] CreateClusterRoleBinding karmada:agent-autoapprove-bootstrap already exists.
I1117 13:23:37.783616  435363 tlsbootstrap.go:60] [bootstrap-token] configured RBAC rules to allow certificate rotation for all agent client certificates in the member cluster
I1117 13:23:37.983712  435363 rbac.go:86] CreateClusterRoleBinding karmada:agent-autoapprove-certificate-rotation already exists.
I1117 13:23:37.984289  435363 deploy.go:126] Initialize karmada bootstrap token
I1117 13:23:37.989430  435363 deploy.go:397] create karmada kube controller manager Deployment
I1117 13:23:37.995075  435363 idempotency.go:215] Service karmada-system/kube-controller-manager has been created or updated.
I1117 13:23:41.004977  435363 check.go:49] pod: kube-controller-manager-69557c876f-f7dn6 is ready. status: Running
I1117 13:23:41.004998  435363 deploy.go:410] create karmada scheduler Deployment
I1117 13:23:44.012973  435363 check.go:49] pod: karmada-scheduler-794dc9f45f-8r4r8 is ready. status: Running
I1117 13:23:44.012994  435363 deploy.go:420] create karmada controller manager Deployment
W1117 13:23:47.020725  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:23:48.023418  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:23:49.022448  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:23:50.023802  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:23:51.023228  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:23:52.022516  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:23:53.022805  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:23:54.022510  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:23:55.022815  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:23:56.022726  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:23:57.022885  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:23:58.022653  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:23:59.022756  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:24:00.023082  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:24:01.023540  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:24:02.022870  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:24:03.023268  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:24:04.022995  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:24:05.022559  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:24:06.022661  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:24:07.023025  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:24:08.022828  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:24:09.023436  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:24:10.022639  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:24:11.022535  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:24:12.023088  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:24:13.022549  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:24:14.022755  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:24:15.022653  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:24:16.022915  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:24:17.022788  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:24:17.024149  435363 check.go:52] pod: karmada-controller-manager-5bcff8fb76-vzxst not ready. status: ContainerCreating
W1117 13:24:17.024170  435363 deploy.go:425] timed out waiting for the condition
I1117 13:24:17.024181  435363 deploy.go:430] create karmada webhook Deployment
I1117 13:24:17.030093  435363 idempotency.go:215] Service karmada-system/karmada-webhook has been created or updated.
I1117 13:24:20.041393  435363 check.go:49] pod: karmada-webhook-67c5ddd6d5-bmbvx is ready. status: Running

------------------------------------------------------------------------------------------------------
 █████   ████   █████████   ███████████   ██████   ██████   █████████   ██████████     █████████
░░███   ███░   ███░░░░░███ ░░███░░░░░███ ░░██████ ██████   ███░░░░░███ ░░███░░░░███   ███░░░░░███
 ░███  ███    ░███    ░███  ░███    ░███  ░███░█████░███  ░███    ░███  ░███   ░░███ ░███    ░███
 ░███████     ░███████████  ░██████████   ░███░░███ ░███  ░███████████  ░███    ░███ ░███████████
 ░███░░███    ░███░░░░░███  ░███░░░░░███  ░███ ░░░  ░███  ░███░░░░░███  ░███    ░███ ░███░░░░░███
 ░███ ░░███   ░███    ░███  ░███    ░███  ░███      ░███  ░███    ░███  ░███    ███  ░███    ░███
 █████ ░░████ █████   █████ █████   █████ █████     █████ █████   █████ ██████████   █████   █████
░░░░░   ░░░░ ░░░░░   ░░░░░ ░░░░░   ░░░░░ ░░░░░     ░░░░░ ░░░░░   ░░░░░ ░░░░░░░░░░   ░░░░░   ░░░░░
------------------------------------------------------------------------------------------------------
Karmada is installed successfully.

Register Kubernetes cluster to Karmada control plane.

Register cluster with 'Push' mode

Step 1: Use "karmadactl join" command to register the cluster to Karmada control plane. --cluster-kubeconfig is kubeconfig of the member cluster.
(In karmada)~# MEMBER_CLUSTER_NAME=$(cat ~/.kube/config  | grep current-context | sed 's/: /\n/g'| sed '1d')
(In karmada)~# karmadactl --kubeconfig /etc/karmada/karmada-apiserver.config  join ${MEMBER_CLUSTER_NAME} --cluster-kubeconfig=$HOME/.kube/config

Step 2: Show members of karmada
(In karmada)~# kubectl --kubeconfig /etc/karmada/karmada-apiserver.config get clusters


Register cluster with 'Pull' mode

Step 1: Use "karmadactl register" command to register the cluster to Karmada control plane. "--cluster-name" is set to cluster of current-context by default.
(In member cluster)~# karmadactl register 10.130.59.18:32443 --token acp5xq.1ifzfakl5t04xx2k --discovery-token-ca-cert-hash sha256:b809ac8fc40ddb1b1aae7df677ef916e54d170c3148c573f123b743b72ca043b

Step 2: Show members of karmada
(In karmada)~# kubectl --kubeconfig /etc/karmada/karmada-apiserver.config get clusters
```
And this is the pod status:
```
 ⚡ root@k8s-master  ~  k get pod -w                                                                                                           [13:22:42 11/17]
NAME     READY   STATUS    RESTARTS   AGE
etcd-0   1/1     Running   0          2s
karmada-apiserver-645bff9854-x2nlk   0/1     Pending   0          0s
karmada-apiserver-645bff9854-x2nlk   0/1     Pending   0          0s
karmada-apiserver-645bff9854-x2nlk   0/1     ContainerCreating   0          0s
karmada-apiserver-645bff9854-x2nlk   0/1     Running             0          1s
karmada-apiserver-645bff9854-x2nlk   1/1     Running             0          30s
karmada-aggregated-apiserver-5f77d747fd-z5c5s   0/1     Pending             0          0s
karmada-aggregated-apiserver-5f77d747fd-z5c5s   0/1     Pending             0          0s
karmada-aggregated-apiserver-5f77d747fd-z5c5s   0/1     ContainerCreating   0          0s
karmada-aggregated-apiserver-5f77d747fd-z5c5s   0/1     Running             0          1s
karmada-aggregated-apiserver-5f77d747fd-z5c5s   1/1     Running             0          2s
kube-controller-manager-69557c876f-f7dn6        0/1     Pending             0          0s
kube-controller-manager-69557c876f-f7dn6        0/1     Pending             0          0s
kube-controller-manager-69557c876f-f7dn6        0/1     ContainerCreating   0          0s
kube-controller-manager-69557c876f-f7dn6        1/1     Running             0          1s
karmada-scheduler-794dc9f45f-8r4r8              0/1     Pending             0          0s
karmada-scheduler-794dc9f45f-8r4r8              0/1     Pending             0          0s
karmada-scheduler-794dc9f45f-8r4r8              0/1     ContainerCreating   0          0s
karmada-scheduler-794dc9f45f-8r4r8              1/1     Running             0          1s
karmada-controller-manager-5bcff8fb76-vzxst     0/1     Pending             0          0s
karmada-controller-manager-5bcff8fb76-vzxst     0/1     Pending             0          0s
karmada-controller-manager-5bcff8fb76-vzxst     0/1     ContainerCreating   0          0s
karmada-webhook-67c5ddd6d5-bmbvx                0/1     Pending             0          0s
karmada-webhook-67c5ddd6d5-bmbvx                0/1     Pending             0          0s
karmada-webhook-67c5ddd6d5-bmbvx                0/1     ContainerCreating   0          0s
karmada-webhook-67c5ddd6d5-bmbvx                0/1     Running             0          1s
karmada-webhook-67c5ddd6d5-bmbvx                1/1     Running             0          1s
karmada-controller-manager-5bcff8fb76-vzxst     1/1     Running             0          51s
```

File: pkg/karmadactl/util/idempotency.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (cleverhu) [reply pkg/karmadactl/util/idempotency.go:92]: > ObjectMeta.Name

I create a pr for this, https://github.com/karmada-io/karmada/pull/2823,PTAL.

File: pkg/karmadactl/util/idempotency.go [hunk 2]
```diff
@@ -88,6 +88,8 @@ func CreateOrUpdateAPIService(apiRegistrationClient *aggregator.Clientset, apise
 			return fmt.Errorf("unable to update APIService: %v", err)
 		}
 	}
+
+	klog.V(2).Infof("apiService %s has been created or updated.", apiservice.Name)
```
Comment (lonelyCZ) [inline pkg/karmadactl/util/idempotency.go:92]:
```suggestion
	klog.V(2).Infof("APIService %s has been created or updated.", apiservice.Name)
```

File: pkg/karmadactl/cmdinit/karmada/deploy.go [hunk 3]
```diff
@@ -298,27 +299,10 @@ func initAPIService(clientSet *kubernetes.Clientset, restConfig *rest.Config, sy
...
-			klog.Infof("Update APIService '%s'", aaAPIServiceObjName)
-			aaAPIService.ObjectMeta.ResourceVersion = v.ResourceVersion
-			if _, err := apiRegistrationClient.ApiregistrationV1().APIServices().Update(context.TODO(), aaAPIService, metav1.UpdateOptions{}); err != nil {
-				return err
-			}
-			return nil
-		}
-	}
-
-	klog.Infof("Create APIService '%s'", aaAPIServiceObjName)
```
Comment (lonelyCZ) [inline pkg/karmadactl/cmdinit/karmada/deploy.go:318]: we can retain this log.

Comment (cleverhu) [reply pkg/karmadactl/cmdinit/karmada/deploy.go:318]: Done.

**Comment by lonelyCZ** [comment]
It works fine in my env.

Thanks @cleverhu 

/lgtm
/approve

**Pull request merged**